clwb/src/com/google/idea/blaze/clwb/run/producers/RunConfigurationProducerSuppressor.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.idea.blaze.clwb.run.producers;
+
+import com.google.idea.blaze.base.settings.Blaze;
+import com.intellij.execution.RunConfigurationProducerService;
+import com.intellij.execution.actions.RunConfigurationProducer;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.startup.StartupActivity;
+import com.jetbrains.cidr.cpp.execution.testing.boost.CMakeBoostTestRunConfigurationProducer;
+import com.jetbrains.cidr.cpp.execution.testing.google.CMakeGoogleTestRunConfigurationProducer;
+import com.jetbrains.cidr.cpp.execution.testing.tcatch.CMakeCatchTestRunConfigurationProducer;
+import org.jetbrains.annotations.NotNull;
+
+/** Suppresses certain non-Blaze configuration producers in Blaze projects. */
+public abstract class RunConfigurationProducerSuppressor implements StartupActivity.Background {
+
+  @NotNull
+  protected abstract Class<? extends RunConfigurationProducer<?>> producerType();
+
+  @Override
+  public void runActivity(@NotNull Project project) {
+    if (Blaze.isBlazeProject(project)) {
+      RunConfigurationProducerService producerService =
+              RunConfigurationProducerService.getInstance(project);
+      producerService.addIgnoredProducer(producerType());
+    }
+  }
+
+  public static class Boost extends RunConfigurationProducerSuppressor {
```

Comment:
To be safe, I'd put each of these class in a separate file (to avoid potential errors during class loading when one of the plugins is disabled).

Reply:
I have tested this by loading a C++ Bazel project with all three plugins disabled, and I didn't see any errors.

In fact, this whole situation is a bit strange, because the only reason why the Bazel plugin needs to have these testing plugins in the classpath is to know the names of the classes that it tries to disable. Since it is only the name of the class that matters, we could could have hardcoded it as a String, but unfortunately `RunConfigurationProducerService` [requires](https://github.com/JetBrains/intellij-community/blob/a39956e5e1712c9bec5b470b2362fef32e711119/platform/lang-api/src/com/intellij/execution/RunConfigurationProducerService.java#L56) a Class reference (only to call getName() on it immediately).

Reply:
Maybe we should go this way and just remove dependencies on these plugins completely?